armory/utils/export.py

```diff
@@ -231,22 +249,83 @@ def get_sample(
             raise TypeError("Both y_i and y_pred are None, but with_boxes is True")
         box_layer = ImageDraw.Draw(image)
 
+        gt_coco_items = []
+        pred_coco_items = []
+
+        if y_i is None or y_i_pred is None:
+            log.warning("Cannot export some coco jsons due to missing data.")
+
         if y_i is not None:
             bboxes_true = y_i["boxes"]
             labels_true = y_i["labels"]
+            try:
+                image_id = y_i["image_id"][0]  # All boxes in y_i are for the same image
+            except Exception:
+                image_id = None
```

Comment:
is there a specific reason for the try/except block for `image_id`? All object detection datasets should have this key. If there is, `except KeyError` would seem more appropriate.

Note: the CARLA video tracking dataset just has a np array for its `y`'s, not a dict with keys including `image_id`.

R.e. the warning message `Cannot export some coco jsons...`, it looks to me like the JSON will still be outputted? I don't see any code preventing that from happening, but perhaps I'm missing it. It also would seem to me like `int(image_id)` would fail later if `image_id` really is `None` (but again, I don' think it ever should be)

Reply:
Line 271/294 says "if image_id is None: continue" so it doesn't add anything to the list of box data (and also won't fail at int(image_id) because it won't get there).  Then the file is only exported if that list is longer than 0.

So yeah I'd assume that most real datasets have that image_id field, but my assumption is naive since I don't know much about coco data.  The sample exporter tests you wrote failed because the fake test didn't have that field.  That's why I added a check.  If you're sure we won't encounter that in real life, we can rewrite the tests and remove that try/except.

Reply:
All object detection datasets should have the `image_id` field, so I'd suggest we remove code that should never be reached (i.e. simply grab `image_id` without a try/except block)

Reply:
Sounds good